src/Build/Resources/Constants.cs

```diff
@@ -365,6 +365,14 @@ private static void InitializeAvailableMethods()
                         availableStaticMethods.TryAdd("Microsoft.Build.Utilities.ToolLocationHelper", new Tuple<string, Type>("Microsoft.Build.Utilities.ToolLocationHelper, Microsoft.Build.Utilities.Core, Version=" + MSBuildConstants.CurrentAssemblyVersion + ", Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", null));
                         availableStaticMethods.TryAdd("System.Runtime.InteropServices.RuntimeInformation", runtimeInformationType);
                         availableStaticMethods.TryAdd("System.Runtime.InteropServices.OSPlatform", osPlatformType);
+#if !NET5_0_OR_GREATER
+                        // Add alternate type for System.OperatingSystem static methods.
+                        var operatingSystemType = new Tuple<string, Type>("Microsoft.Build.Framework.OperatingSystem, Microsoft.Build.Framework, Version=" + MSBuildConstants.CurrentAssemblyVersion + ", Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", null);
+                        availableStaticMethods.TryAdd("System.OperatingSystem", operatingSystemType);
+                        availableStaticMethods.TryAdd("Microsoft.Build.Framework.OperatingSystem", operatingSystemType);
```

Comment:
why do we need to register this type?

Reply:
Do you mean to ask why are there two entries? I may have missed something but when stepping through the code for processing a property function, there are two lookups performed against `s_availableStaticMethods`.

The first lookup gets the receiver type using `System.OperatingSystem` as the key, effectively mapping `System.OperatingSystem` to `Microsoft.Build.Framework.OperatingSystem`.

The second lookup uses the receiver type, i.e. `Microsoft.Build.Framework.OperatingSystem`, as the key.

Reply:
Hmm shouldn't the first lookup be enough though?

Reply:
Apparently not. If I remove the second entry, the property function resolution fails with an error that `Microsoft.Build.Framework.OperatingSystem` is not available.

Reply:
ok, thanks for checking